src/events/parser.rs

```diff
@@ -0,0 +1,280 @@
+use anyhow::Result;
+use serde_json;
+use serde_yaml;
+use std::collections::BTreeMap;
+use std::io::BufReader;
+use std::path::PathBuf;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum ComponentType {
+    actor,
+    capability,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Annotations {
+    version: String,
+    description: String,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Metadata {
+    name: String,
+    annotations: Annotations,
+}
+
+fn default_link_name() -> Option<String> {
+    Some("default".to_string())
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Properties {
+    image: String,
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    contract: Option<String>,
+    #[serde(default = "default_link_name", skip_serializing_if = "Option::is_none")]
+    link_name: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub enum TraitType {
+    spreadscaler,
+    linkdef,
+}
+#[derive(Debug, Serialize, Deserialize)]
+pub enum SpreadRequirements {
```

Comment:
So the `requirements` object is actually just a map of string, string because they can be entirely arbitrary. So all you should need for `requirements` is `BTreeMap<String, String>` rather than `BTreeMap<String, SpreadRequirementValues>`

Reply:
Putting the value as String fails because if you checkout [this](https://github.com/wasmCloud/wadm/blob/main/oam/simple1.yaml) manifest on line 53, the value is treated as a boolean instead of String. Which is why the Map needs to accept a variety of values.

Reply:
I think this is a case where we can break with what wadm is currently doing. A map of string, string is what host labels are and that is what we are currently matching. Plus it makes the code much easier. So I think it'd be ok to change